cache/lru/cache.go

```diff
@@ -56,14 +67,23 @@ func (c *Cache[K, _]) Evict(key K) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
-	c.elements.Delete(key)
+	c.evict(key)
+}
+
+func (c *Cache[K, V]) evict(key K) {
+	if value, ok := c.elements.Get(key); ok {
+		c.onEvict(key, value)
+		c.elements.Delete(key)
+	}
```

Comment:
nit: early return where possible for readability
```suggestion
	value, ok := c.elements.Get(key)
	if !ok {
		return
	}
	
	c.onEvict(key, value)
	c.elements.Delete(key)
```